### DISCUSSION TITLE
Nested Stacks: TemplateURL vs templateFile

### DISCUSSION THREAD

**Discussion by iamcrisb** [aws/aws-cdk#23740] [Q&A]
Hi, I was reading the [docs](https://docs.aws.amazon.com/cdk/v2/guide/use_cfn_template.html) for creating nested stacks that are defined in my main template and I can not figure out why do we need to specify the templateFile by code and save it locally as long as a TemplateURL is being provided in the main template.

<img width="976" alt="image" src="https://user-images.githubusercontent.com/41893422/213302234-8160b6e2-86c8-423a-b938-f358c615c8b6.png">

What the purpose of having the TemplateUrl if it is not being used at all?

**Comment by khushail** [comment]
Hi @cristianbuta , when you define a nested stack as a separate stack construct, the `templatefile` property specifies the file in your local system which means, AWS CDK uploads this to S3 Bucket and uses this to generate the `templateURL`property in the main template. So while `templateURL` property is not directly used, its still used to deploy the stack and to reference the the nested stack's cloudformation template. Hope this answers your question. let me know if there are any further questions.

**Reply by iamcrisb** [reply]
Thank you for your answer, so based on this I should be able to omit the templateFile property so the templateURL gets picked up automatically, but cdk does not allow this, what am I missing?

**Comment by khushail** [comment]
Here is the [documentation](https://docs.aws.amazon.com/cdk/v2/guide/use_cfn_template.html#use_cfn_template_cfninclude_nested) that explains use of both methods.
